salma-hayek/src/e/util/ProcessUtilities.java

```diff
@@ -93,31 +95,27 @@ public static int backQuote(final File directory, final String[] commandArray, f
      * 
      * If directory is null, the subprocess inherits our working directory.
      */
-    public static int runCommand(final File directory, final String[] commandArray, final ProcessListener processListener, final String input, final LineListener outputLineListener, final LineListener errorLineListener) {
+    public static int runCommand(final Path directory, final String[] commandArray, final ProcessListener processListener, final String input, final LineListener outputLineListener, final LineListener errorLineListener) {
         List<String> command = Arrays.asList(commandArray);
         int status = 1; // FIXME: should we signal "something went wrong" more distinctively?
         try {
             final Process p = exec(command, directory);
             if (processListener != null) {
                 processListener.processStarted(p);
             }
-            Thread inputWriterThread = new Thread(new Runnable() {
-                public void run() {
-                    try {
-                        BufferedWriter out = new BufferedWriter(new OutputStreamWriter(p.getOutputStream(), "UTF-8"));
-                        out.append(input);
-                        out.flush();
-                        out.close();
-                    } catch (Exception ex) {
-                        feedExceptionToLineListener(errorLineListener, ex);
-                    }
+            Thread inputWriterThread = new Thread(() -> {
+                try {
+                    BufferedWriter out = new BufferedWriter(new OutputStreamWriter(p.getOutputStream(), "UTF-8"));
```

Comment:
should probably use try-with-resources to ensure the OutputStream is closed even when an exception is encountered